### ISSUE TITLE
🐛 post-removal script is not cleaning everything

### ISSUE THREAD

**Issue by eximius313** [cloudflare/cloudflared#686]
**Describe the bug**
Installation is successful, but the purging (`dpkg --purge cloudflared`) does not clean the service
```
$ ls /etc/systemd/system/cloudflared*
cloudflared.service 
cloudflared-update.service  
cloudflared-update.timer
```

nor config files:
```
$ ls /etc/cloudflared/
config.yml
```

therefore purge shows following errors:
```
dpkg: error processing package cloudflared (--purge):
 installed cloudflared package post-removal script subprocess returned error exit status 1
```

and after reinstall (`apt-get install --reinstall cloudflared`) when one tries to install new service, they get:
```
cloudflared service is already installed at /etc/systemd/system/cloudflared.service; if you are running a cloudflared tunnel, you can point it to multiple origins, avoiding the need to run more than one cloudflared service in the same machine; otherwise if you are really sure, you can do `cloudflared service uninstall` to clean up the existing service and then try again this command
```

Moreover - I was getting weird errors while trying to `systemctl disable cloudflared`:
```
Unit /etc/systemd/system/cloudflared.service is masked, ignoring.
```

**To Reproduce**
1. `dpkg --purge cloudflared`

**Expected behavior**
`post-removal script` should clean everything.

**Environment and versions**
 - OS: Linux
 - Architecture: Arm64
 - Version: 2022.6.3

**Comment by Soulsender** [comment]
Had this same issue, this solution worked for me:

```sudo mv /var/lib/dpkg/info/cloudflared.* /tmp/
sudo dpkg --remove --force-remove-reinstreq cloudflared
sudo apt-get remove cloudflared
sudo apt-get autoremove && sudo apt-get autoclean
```

**Comment by fabiannagel** [comment]
+1, still an issue

**Comment by rubenz91** [comment]
Same here, mine kind of crashed, then after doing the official step to uninstall left garbage that dint allow to uninstall and now its even worse as states the follow doing the cleanups: 

`E: The package cloudflared needs to be reinstalled, but I can't find an archive for it.
`

Fir the dpkg:

`sudo dpkg --remove --force-remove-reinstreq cloudflared
dpkg: warning: overriding problem because --force enabled:
dpkg: warning: package is in a very bad inconsistent state; you should
 reinstall it before attempting a removal
(Reading database ... 222495 files and directories currently installed.)
Removing cloudflared (2023.7.3) ...
rm: cannot remove '/usr/local/bin/cloudflared': No such file or directory
dpkg: error processing package cloudflared (--remove):
 installed cloudflared package post-removal script subprocess returned error exit status 1
Errors were encountered while processing:
 cloudflared
`

**Comment by rubenz91** [comment]
> Same here, mine kind of crashed, then after doing the official step to uninstall left garbage that dint allow to uninstall and now its even worse as states the follow doing the cleanups:
> 
> `E: The package cloudflared needs to be reinstalled, but I can't find an archive for it. `
> 
> For the dpkg:
> 
> `sudo dpkg --remove --force-remove-reinstreq cloudflared dpkg: warning: overriding problem because --force enabled: dpkg: warning: package is in a very bad inconsistent state; you should reinstall it before attempting a removal (Reading database ... 222495 files and directories currently installed.) Removing cloudflared (2023.7.3) ... rm: cannot remove '/usr/local/bin/cloudflared': No such file or directory dpkg: error processing package cloudflared (--remove): installed cloudflared package post-removal script subprocess returned error exit status 1 Errors were encountered while processing: cloudflared `

At the end I found this [post](https://www.reddit.com/r/CloudFlare/comments/vcz5cz/remove_cloudflare_from_ubuntu_2204/) and seems the section that worked was the located and delete everything there:
` locate cloudflare
  sudo rm -rf [put each of the file paths in here one by one unless you can shorted them with a *]
`

Also I removed certbot that I had thats the only steps I did and it allow me to reinstall it.

**Comment by BastianZim** [comment]
Also getting this issue with cloudflare on Raspberry Pi

**Comment by b4shful** [comment]
The service still doesn't get removed when uninstalling cloudflared (I used `sudo apt remove cloudflared`), which causes a lot of errors - just discovered this looping over and over again on one of my systems...
```
Jan 22 16:41:35 lobster systemd[1]: Starting cloudflared...
Jan 22 16:41:35 lobster systemd[55475]: cloudflared.service: Failed to locate executable /usr/bin/cloudflared: No such file or directory
Jan 22 16:41:35 lobster systemd[55475]: cloudflared.service: Failed at step EXEC spawning /usr/bin/cloudflared: No such file or directory
Jan 22 16:41:35 lobster systemd[1]: cloudflared.service: Main process exited, code=exited, status=203/EXEC
Jan 22 16:41:35 lobster systemd[1]: cloudflared.service: Failed with result 'exit-code'.
Jan 22 16:41:35 lobster systemd[1]: Failed to start cloudflared.
Jan 22 16:41:40 lobster systemd[1]: cloudflared.service: Scheduled restart job, restart counter is at 36562.
Jan 22 16:41:40 lobster systemd[1]: Stopped cloudflared.
Jan 22 16:41:40 lobster systemd[1]: Starting cloudflared...
Jan 22 16:41:40 lobster systemd[55483]: cloudflared.service: Failed to locate executable /usr/bin/cloudflared: No such file or directory
Jan 22 16:41:40 lobster systemd[55483]: cloudflared.service: Failed at step EXEC spawning /usr/bin/cloudflared: No such file or directory
Jan 22 16:41:40 lobster systemd[1]: cloudflared.service: Main process exited, code=exited, status=203/EXEC
Jan 22 16:41:40 lobster systemd[1]: cloudflared.service: Failed with result 'exit-code'.
Jan 22 16:41:40 lobster systemd[1]: Failed to start cloudflared.
Jan 22 16:41:45 lobster systemd[1]: cloudflared.service: Scheduled restart job, restart counter is at 36563.
Jan 22 16:41:45 lobster systemd[1]: Stopped cloudflared.
```
Running `sudo apt purge cloudflared` then causes errors:
```
Purging configuration files for cloudflared (2024.1.4) ...
rm: cannot remove '/usr/local/bin/cloudflared': No such file or directory
dpkg: error processing package cloudflared (--purge):
 installed cloudflared package post-removal script subprocess returned error exit status 1
Errors were encountered while processing:
 cloudflared
E: Sub-process /usr/bin/dpkg returned an error code (1)
```
Installing it and attempting removal again with `sudo apt purge cloudflared` leads to:
```
(Reading database ... 77149 files and directories currently installed.)
Removing cloudflared (2024.1.4) ...
Processing triggers for man-db (2.10.2-1) ...
(Reading database ... 77145 files and directories currently installed.)
Purging configuration files for cloudflared (2024.1.4) ...
rm: cannot remove '/usr/local/bin/cloudflared': No such file or directory
dpkg: error processing package cloudflared (--purge):
 installed cloudflared package post-removal script subprocess returned error exit status 1
Errors were encountered while processing:
 cloudflared
E: Sub-process /usr/bin/dpkg returned an error code (1)
```
Pretty terrible problem :(

**Comment by BastianZim** [comment]
Yeah is there any chance we can get the priority escalated? This is pretty major.

**Comment by b4shful** [comment]
@chungthuang apologies for the ping, but the user assigned to this (sudarshan-reddy) has "previously helped build cloudflare tunnels" in their bio and hasn't committed anything since October. So they may be no longer working on this...

I found your username by looking through recent commits.

Could you take a look and/or assign someone?

**Comment by joej** [comment]
This exact thing is happening in my reinstallation -- 04/13/2024.
Why, in any semblance of sanity, is this still happening since before 2022?

**Comment by b4shful** [comment]
@joej I can't reproduce this on Ubuntu 22.04, to test I just installed, configured, then removed and reinstalled on my side and it worked fine.
Do make sure you follow the instructions [here](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/get-started/create-local-tunnel/) if on Linux to add cloudflare's apt repository to your system, so that you get the latest version (this fix was only deployed in versions 2024.3.0 and above - `cloudflared --version` can confirm this but of course that may not work if you are having reinstallation issues).

Feel free to post some more information about your setup, what steps you took to uninstall/reinstall, and whether you are sure you are using an updated version of `cloudflared` that has this fix (#1166) applied.
